sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -20,21 +25,110 @@
      */
     XML;
 
+    private static final ClientLogger LOGGER = new ClientLogger(SerializerEncoding.class);
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final List<SerializerEncodingMimeTypes> SUPPORTED_MIME_TYPES;
+    private static final SerializerEncoding DEFAULT_ENCODING = JSON;
+
+    static {
+        SUPPORTED_MIME_TYPES = new ArrayList<>();
```

Comment:
Thoughts on making this more simplified using a case insensitive `TreeMap`. Begin by adding in well known Content-Types such as `application/json`, `text/xml`, and `application/xml` (plus additional in the future as we expand), then add in a simple `"json" -> SerializerEncoding.JSON` and same for XML.

During the conversion from `HttpHeaders` to `SerializerEncoding` we use the following checks:

1. Missing ContentType or it is null or empty, log a warning and return JSON.
2. Check if the ContentType exists in the `TreeMap`, if it does return that value.
3. Use string splitting logic, finding the sub-type and if it exists the `+` split.

This would simplify out a lot of the class overhead and generally have the same result. Additionally, the map could be updated with the Content-Types that weren't in the well known set to optimize them in future cases when they appear (this runs a risk of having many Content-Types if they get randomized a lot but I think that would be a bad design).

Reply:
Another option instead of `TreeMap` is `ConcurrentSkipListMap` which also has the option for a key comparator.